### ISSUE TITLE
Refactoring galore

### ISSUE THREAD

**Issue by jchio001** [kkong006/CS180FrontEnd#4]
So now that I'm free from the shackles of school, I will not be satisfied with "code looks ok and stuff works good enough", and I will personally comb through all lines of code.

Major thing to overhaul is that I'm noticing for some listviews, the data the adapters is using is being stored  in a brazillion arrays. This doesn't make any sense and it wastes processing power/time/memory, since you can just have the adapter take in an array of objects, and then pass it in the list of objects obtained from the Retrofit response vs parsing the list of objects into a bunch of arrays and passing that into the adapter. Don't need to think too far for adapters, it's pretty much plug in array of objects into adapter and stuff gets handled. No parsing needed.

That's the main thing I've noticed in terms of code cleanliness; find all occurrences the aforementioned adapter stuff and fix it. I've touched up an adapter somewhere, and you can use that as the basis.

tl;dr clean adapters; this ticket is free for anyone to pick up.

**Comment by jchio001** [comment]
Code refactoring never ends.

As of now, need to refactor how the app handles an empty feed.
